moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryProjectEntity.js

```diff
@@ -18,38 +21,59 @@ import { PROJECT_UPDATE_SPONSOR } from "../../../../queries/project";
  * @returns {JSX.Element}
  * @constructor
  */
-const ProjectSummaryProjectSponsor = ({
+const ProjectSummaryProjectEntity = ({
   projectId,
   data,
   refetch,
   classes,
   snackbarHandle,
+  entityName,
 }) => {
   const entityList = data?.moped_entity ?? [];
-  const noneSponsor = entityList.find(e => e.entity_id === 0);
-  const originalSponsor = entityList.find(
-    e => e.entity_id === data?.moped_project?.[0]?.project_sponsor
-  );
+  const noneSponsor = entityList.find((e) => e.entity_id === 0);
+
+  const getOriginalSponsor = () => {
+    if (entityName === "Lead") {
+      return entityList.find(
+        (e) => e.entity_id === data?.moped_project?.[0]?.project_lead_id
+      );
+    }
+    if (entityName === "Sponsor") {
+      return entityList.find(
+        (e) => e.entity_id === data?.moped_project?.[0]?.project_sponsor
+      );
```

Comment:
this isn't a comment on your implementation of this, because you didnt start the pattern that we have all over moped, but we could leverage the object relationships set up in hasura to get the entity name out of the query instead doing a lookup with in the component.

Reply:
@chiaberry you're right, @johnclary mentioned using a hasura object relationship in the issue and i glossed over that part. i'm going to look into how to implement that now!